src/arraymancer/laser/tensor/datatypes.nim

```diff
@@ -165,43 +172,43 @@ func unsafe_raw_buf*[T: KnownSupportsCopyMem](t: var Tensor[T], aligned: static
   ## while the pointer is valid, all data accesses will be through it (no aliasing)
   ## and that the data is aligned by LASER_MEM_ALIGN (default 64).
-  unsafe_raw_offset_impl(t.offset)
+  unsafe_raw_offset_impl(t.offset+offset)
 
-func unsafe_raw_offset*[T: KnownSupportsCopyMem](t: var Tensor[T], aligned: static bool = true): RawMutableView[T] {.inline.} =
+func unsafe_raw_offset*[T: KnownSupportsCopyMem](t: var Tensor[T], offset:int = 0, aligned: static bool = true): RawMutableView[T] {.inline.} =
   ## Returns a view to the start of the valid data
   ##
   ## Unsafe: the pointer can outlive the input tensor
   ## For optimization purposes, Laser will hint the compiler that
   ## while the pointer is valid, all data accesses will be through it (no aliasing)
   ## and that the data is aligned by LASER_MEM_ALIGN (default 64).
-  unsafe_raw_offset_impl(t.offset)
+  unsafe_raw_offset_impl(t.offset+offset)
 
 func unsafe_raw_buf*[T: not KnownSupportsCopyMem](t: Tensor[T], aligned: static bool = true): ptr UncheckedArray[T]  {.error: "Access via raw pointer forbidden for non mem copyable types!".}
 
 func unsafe_raw_offset*[T: not KnownSupportsCopyMem](t: Tensor[T], aligned: static bool = true): ptr UncheckedArray[T] {.error: "Access via raw pointer forbidden for non mem copyable types!".}
 
-macro raw_data_unaligned*(body: untyped): untyped =
-  ## Within this code block, all raw data accesses will not be
-  ## assumed aligned by default (LASER_MEM_ALIGN is 64 by default).
-  ## Use this when interfacing with external buffers of unknown alignment.
-  ##
-  ## ⚠️ Warning:
-  ##     At the moment Nim's builtin term-rewriting macros are not scoped.
-  ##     All processing within the file this is called will be considered
-  ##     unaligned. https://github.com/nim-lang/Nim/issues/7214#issuecomment-431567894.
-  block:
-    template trmUnsafeRawBuf{unsafe_raw_buf(x, aligned)}(x, aligned): auto =
-      {.noRewrite.}: unsafe_raw_buf(x, false)
-    template trmUnsafeRawOffset{unsafe_raw_offset(x, aligned)}(x, aligned): auto =
-      {.noRewrite.}: unsafe_raw_offset(x, false)
-    body
+# macro raw_data_unaligned*(body: untyped): untyped =
+#   ## Within this code block, all raw data accesses will not be
+#   ## assumed aligned by default (LASER_MEM_ALIGN is 64 by default).
+#   ## Use this when interfacing with external buffers of unknown alignment.
+#   ##
+#   ## ⚠️ Warning:
+#   ##     At the moment Nim's builtin term-rewriting macros are not scoped.
+#   ##     All processing within the file this is called will be considered
+#   ##     unaligned. https://github.com/nim-lang/Nim/issues/7214#issuecomment-431567894.
+#   block:
+#     template trmUnsafeRawBuf{unsafe_raw_buf(x, aligned)}(x, aligned): auto =
+#       {.noRewrite.}: unsafe_raw_buf(x, false)
+#     template trmUnsafeRawOffset{unsafe_raw_offset(x, aligned)}(x, aligned): auto =
+#       {.noRewrite.}: unsafe_raw_offset(x, false)
+#     body
```

Comment:
Do we leave this like this? 

As far as I understand it's used nowhere, but then again term rewrite macros are weird. @mratsim?

Reply:
Well  I commented it exaclty for this purpose. To see if stuff breaks; evetnaully dead code should be removed IMO but macros are so hard to test that I wasn't confident enough to remove it (it's easier to un comment code than to go through git history)

Reply:
tbh I've been running with it commented out for months now locally and never had any issues.

Reply:
I do not mind either way, I'll let @mratsim decide if the code should stay or not :)

Reply:
It should be used when stuff from Numpy, or Apache Arrow or Tensorflow or Pytorch is passed to Arraymancer, so when memowner is set to false.